src/main/java/com/ericsson/ei/waitlist/WaitListStorageHandler.java

```diff
@@ -63,15 +64,17 @@
     @Autowired
     private JmesPathInterface jmesPathInterface;
 
-    public void addEventToWaitList(String event, RulesObject rulesObject) throws Exception {
-        String condition = "{\"_id\" : \"" + new JSONObject(event).getJSONObject("meta").getString("id") + "\"}";
-        List<String> foundEventsInWaitList = mongoDbHandler.find(databaseName, collectionName, condition);
-        if (foundEventsInWaitList.isEmpty()) {
-            String input = addPropertiesToEvent(event, rulesObject);
-            boolean result = mongoDbHandler.insertDocument(databaseName, collectionName, input);
-            if (!result) {
-                throw new Exception("Failed to insert the document into database");
+    public void addEventToWaitList(String event, RulesObject rulesObject) {
+        try {
+            String condition = "{\"_id\" : \"" + new JSONObject(event).getJSONObject("meta").getString("id") + "\"}";
+            List<String> foundEventsInWaitList = mongoDbHandler.find(databaseName, collectionName, condition);
+            if (foundEventsInWaitList.isEmpty()) {
+                String input = addPropertiesToEvent(event, rulesObject);
+                mongoDbHandler.insertDocument(databaseName, collectionName, input);
```

Comment:
Move try catch to it´s own function and leave all not mongo related in this function.

Reply:
I agreee, this class could also need some improvemens, but as I said in previous comment I'll try to limit the scope of this PR so will not focus on this class for now. That will be in a separate PR I believe :)